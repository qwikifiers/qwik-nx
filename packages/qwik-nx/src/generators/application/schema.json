{
  "$schema": "http://json-schema.org/schema",
  "$id": "QwikNx",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the application.",
      "pattern": "^[a-zA-Z][^:]*$",
      "x-priority": "important"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "$default": { "$source": "argv", "index": 0 },
      "x-prompt": "Which directory do you want to create the application in?"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          }
        ],
        "default": "css"
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "strict": {
      "type": "boolean",
      "description": "Creates an application with strict mode and strict type checking.",
      "default": true
    },
    "tailwind": {
      "description": "Setup Tailwind",
      "type": "boolean",
      "default": false
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "none"
    },
    "devServerPort": {
      "type": "number",
      "description": "The port for the dev server of the app.",
      "default": 5173
    },
    "previewServerPort": {
      "type": "number",
      "description": "The port for the preview server of the app.",
      "default": 4173
    }
  },
  "required": ["directory"]
}
