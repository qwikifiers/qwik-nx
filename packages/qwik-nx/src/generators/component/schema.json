{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Component",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested)."
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          },
          {
            "value": "none",
            "label": "none"
          }
        ],
        "default": "css"
      }
    },
    "skipTests": {
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "type": "boolean"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    },
    "generateStories": {
      "description": "Create Storybook stories for the component",
      "type": "boolean"
    },
    "exportDefault": {
      "description": "Whether to use default export instead of named one for the generated component.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["name", "project"]
}
