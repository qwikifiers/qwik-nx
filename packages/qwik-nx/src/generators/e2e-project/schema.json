{
  "$schema": "http://json-schema.org/schema",
  "$id": "E2eProject",
  "title": "Create an E2E app for a Qwik app",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "The name of the frontend project to test."
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "playwright"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project", "e2eTestRunner"]
}
