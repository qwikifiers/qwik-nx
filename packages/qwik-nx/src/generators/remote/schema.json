{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Remote",
  "title": "Generate Micro Frontends Setup for Qwik Remote App",
  "description": "Create Micro Frontends configuration files for the Qwik Application.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          }
        ],
        "default": "css"
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "strict": {
      "type": "boolean",
      "description": "Creates an application with strict mode and strict type checking.",
      "default": true
    },
    "tailwind": {
      "description": "Setup Tailwind",
      "type": "boolean",
      "default": false
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "none"
    },
    "port": {
      "type": "number",
      "description": "The port for the dev and preview servers of the app (they're the same by default in MF setup).",
      "default": 4200
    },
    "host": {
      "type": "string",
      "description": "The host / shell application for this remote.",
      "x-priority": "important",
      "x-dropdown": "project"
    }
  },
  "required": ["name"]
}
