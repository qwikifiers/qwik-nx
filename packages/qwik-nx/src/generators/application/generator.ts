import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
