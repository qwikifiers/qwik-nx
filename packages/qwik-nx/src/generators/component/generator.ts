import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  getProjects,
  joinPathFragments,
  logger,
  names,
  Tree,
} from '@nrwl/devkit';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { ComponentGeneratorSchema } from './schema';

interface NormalizedSchema extends ComponentGeneratorSchema {
  directory: string;
  hasStyles: boolean;
  projectRoot: string;
}

function getDirectory(host: Tree, options: ComponentGeneratorSchema) {
  const workspace = getProjects(host);
  let baseDir: string;
  if (options.directory) {
    baseDir = options.directory;
  } else {
    baseDir =
      workspace.get(options.project)!.projectType === 'application'
        ? 'components'
        : 'lib';
  }
  return options.flat
    ? baseDir
    : joinPathFragments(baseDir, names(options.name).fileName);
}

function normalizeOptions(
  host: Tree,
  options: ComponentGeneratorSchema
): NormalizedSchema {
  const project = getProjects(host).get(options.project);

  if (!project) {
    logger.error(
      `Cannot find the ${options.project} project. Please double check the project name.`
    );
    throw new Error();
  }

  const projectRoot =
    project.sourceRoot ?? joinPathFragments(project.root, 'src');

  const directory = getDirectory(host, options);

  return {
    ...options,
    directory,
    hasStyles: options.style !== 'none',
    projectRoot,
  };
}

function createComponentFiles(tree: Tree, options: NormalizedSchema) {
  const libNames = names(options.name);
  const hasStyles = options.style && options.style !== 'none';
  const templateOptions = {
    ...options,
    ...libNames,
    hasStyles,
  };

  const componentDir = joinPathFragments(
    options.projectRoot,
    options.directory
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/common'),
    componentDir,
    templateOptions
  );
  if (hasStyles) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/styles'),
      componentDir,
      templateOptions
    );
  }
  if (!options.skipTests) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/tests'),
      componentDir,
      templateOptions
    );
  }
}

export async function componentGenerator(
  tree: Tree,
  options: ComponentGeneratorSchema
): Promise<GeneratorCallback> {
  const normalizedOptions = normalizeOptions(tree, options);
  createComponentFiles(tree, normalizedOptions);
  await formatFiles(tree);

  return addStyledModuleDependencies(tree, normalizedOptions.style);
}

export default componentGenerator;
