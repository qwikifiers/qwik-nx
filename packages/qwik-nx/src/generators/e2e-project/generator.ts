import {
  ensurePackage,
  GeneratorCallback,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import { getInstalledNxVersion } from '../../utils/get-installed-nx-version';
import { nxKitVersion } from '../../utils/versions';
import { E2eProjectGeneratorSchema, NormalizedSchema } from './schema';
import { normalizeOptions } from './utils/normalize-options';

export async function addE2eProject(
  tree: Tree,
  options: E2eProjectGeneratorSchema
): Promise<GeneratorCallback> {
  const projectConfiguration = readProjectConfiguration(tree, options.project);

  if (projectConfiguration.projectType !== 'application') {
    throw new Error('Cannot setup e2e project for the given frontend project.');
  }

  const normalizedOptions = normalizeOptions(tree, options);
  if (options.e2eTestRunner === 'cypress') {
    return addCypress(tree, normalizedOptions);
  }

  if (options.e2eTestRunner === 'playwright') {
    return addPlaywright(tree, normalizedOptions);
  }

  return () => void 0;
}

async function addCypress(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nrwl/cypress', getInstalledNxVersion(tree));
  const { cypressProjectGenerator } = await import('@nrwl/cypress');

  return await cypressProjectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    project: options.project,
    bundler: 'vite',
    skipFormat: options.skipFormat,
  });
}

async function addPlaywright(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nxkit/playwright', nxKitVersion);
  const { projectGenerator } = await import('@nxkit/playwright');

  return await projectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    frontendProject: options.project,
    skipFormat: options.skipFormat,
  });
}

export default addE2eProject;
