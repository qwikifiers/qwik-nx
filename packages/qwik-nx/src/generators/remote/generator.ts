import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  logger,
  readJson,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { addMicroFrontendBetaWarning } from '../../utils/mf-beta-warning';
import appGenerator from '../application/generator';
import {
  NormalizedSchema,
  QwikAppGeneratorSchema,
} from '../application/schema';
import { normalizeOptions } from '../application/utils/normalize-options';
import { RemoteGeneratorSchema } from './schema';

export async function remoteGenerator(
  tree: Tree,
  options: RemoteGeneratorSchema
) {
  addMicroFrontendBetaWarning();
  const tasks: GeneratorCallback[] = [];

  const appGeneratorSchema: QwikAppGeneratorSchema = {
    ...options,
    devServerPort: options.port,
    previewServerPort: options.port,
    skipFormat: true,
  };
  const initTask = await appGenerator(tree, appGeneratorSchema);
  tasks.push(initTask);

  const normalizedSchema = normalizeOptions(tree, appGeneratorSchema);

  updateFiles(tree, normalizedSchema);

  if (options.host) {
    updateHostWithRemoteConfig(tree, options.host, normalizedSchema);
  }

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
  return runTasksInSerial(...tasks);
}

export default remoteGenerator;

function updateHostWithRemoteConfig(
  tree: Tree,
  hostProjectName: string,
  schema: NormalizedSchema
) {
  const hostConfig = readProjectConfiguration(tree, hostProjectName);
  const remoteConfigPath = joinPathFragments(
    hostConfig.root,
    'src/config/remotes.json'
  );

  if (tree.exists(remoteConfigPath)) {
    const config = readJson(tree, remoteConfigPath);
    config[schema.projectName] = `http://localhost:${schema.devServerPort}`;

    tree.write(remoteConfigPath, JSON.stringify(config));
  } else {
    logger.warn(
      `Could not find remote config at ${remoteConfigPath}. Did you generate this project with "qwik-nx:host"?`
    );
  }
}

function updateFiles(tree: Tree, schema: NormalizedSchema) {
  // TODO: switch to empty preset once available
  tree.delete(joinPathFragments(schema.projectRoot, 'src/routes/flower'));
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/components/header/header.css')
  );
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/components/header/header.tsx')
  );
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/components/icons/qwik.tsx')
  );
  tree.delete(
    joinPathFragments(
      schema.projectRoot,
      'src/components/router-head/router-head.tsx'
    )
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    schema.projectRoot,
    {
      name: schema.projectName,
    }
  );
}
