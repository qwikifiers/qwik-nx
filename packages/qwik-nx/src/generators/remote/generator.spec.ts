import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree, readProjectConfiguration, readJson } from '@nx/devkit';

import { remoteGenerator } from './generator';
import { hostGenerator } from './../host/generator';
import { RemoteGeneratorSchema } from './schema';
import { getFormattedListChanges } from '../../utils/testing-generators';

describe('remote generator', () => {
  let appTree: Tree;
  const options: RemoteGeneratorSchema = { name: 'myremote' };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should run successfully', async () => {
    await remoteGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, 'myremote');
    expect(config).toBeDefined();
    expect(
      appTree.read('apps/myremote/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      appTree.read('apps/myremote/project.json', 'utf-8')
    ).toMatchSnapshot();
    expect(getFormattedListChanges(appTree)).toMatchSnapshot();
  });

  it('should update host config', async () => {
    await hostGenerator(appTree, { name: 'myhost' });

    expect(readJson(appTree, 'apps/myhost/src/config/remotes.json')).toEqual(
      {}
    );

    await remoteGenerator(appTree, { ...options, host: 'myhost', port: 777 });

    expect(readJson(appTree, 'apps/myhost/src/config/remotes.json')).toEqual({
      myremote: 'http://localhost:777',
    });
  });
});
