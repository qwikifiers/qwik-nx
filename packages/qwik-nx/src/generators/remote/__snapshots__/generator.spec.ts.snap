// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remote generator should run successfully 1`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/myremote',
  plugins: [
    qwikNxVite(),
    qwikCity(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myremote/client',
      },
      ssr: {
        outDir: '../../dist/apps/myremote/server',
      },
    }),
    tsconfigPaths({ root: '../../' }),
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`remote generator should run successfully 2`] = `
"{
  \\"name\\": \\"myremote\\",
  \\"$schema\\": \\"../../node_modules/nx/schemas/project-schema.json\\",
  \\"projectType\\": \\"application\\",
  \\"sourceRoot\\": \\"apps/myremote/src\\",
  \\"targets\\": {
    \\"build\\": {
      \\"executor\\": \\"qwik-nx:build\\",
      \\"options\\": {
        \\"runSequence\\": [\\"myremote:build.client\\", \\"myremote:build.ssr\\"],
        \\"outputPath\\": \\"dist/apps/myremote\\"
      },
      \\"configurations\\": {
        \\"preview\\": {}
      }
    },
    \\"build.client\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/myremote\\",
        \\"configFile\\": \\"apps/myremote/vite.config.ts\\"
      }
    },
    \\"build.ssr\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"defaultConfiguration\\": \\"preview\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/myremote\\"
      },
      \\"configurations\\": {
        \\"preview\\": {
          \\"ssr\\": \\"src/entry.preview.tsx\\",
          \\"mode\\": \\"production\\"
        }
      }
    },
    \\"preview\\": {
      \\"executor\\": \\"@nrwl/vite:preview-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"myremote:build\\",
        \\"port\\": 4300
      }
    },
    \\"test\\": {
      \\"executor\\": \\"@nrwl/vite:test\\",
      \\"outputs\\": [\\"../../coverage/apps/myremote\\"],
      \\"options\\": {
        \\"passWithNoTests\\": true,
        \\"reportsDirectory\\": \\"../../coverage/apps/myremote\\"
      }
    },
    \\"serve\\": {
      \\"executor\\": \\"@nrwl/vite:dev-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"myremote:build.client\\",
        \\"mode\\": \\"ssr\\",
        \\"port\\": 4200
      }
    },
    \\"serve.debug\\": {
      \\"executor\\": \\"nx:run-commands\\",
      \\"options\\": {
        \\"command\\": \\"node --inspect-brk ../../node_modules/vite/bin/vite.js --mode ssr --force\\",
        \\"cwd\\": \\"apps/myremote\\"
      }
    }
  },
  \\"tags\\": []
}
"
`;

exports[`remote generator should run successfully 3`] = `
Array [
  Object {
    "path": ".prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/.prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/public/favicon.svg",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/public/manifest.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/public/robots.txt",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/components/.gitkeep",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/components/header/header.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/entry.dev.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/entry.preview.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/entry.ssr.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/global.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/routes/about/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/routes/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/routes/layout.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/src/routes/service-worker.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/tsconfig.app.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myremote/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;
